app/services/ckan/v26/version_diff.rb

```diff
@@ -20,25 +22,28 @@ def call
 
       def diff_create(packages, datasets)
         dataset_uuids = Set[*datasets.pluck(:uuid)]
-        packages.reject { |package| dataset_uuids.include?(package["id"]) }
+
+        packages.reject do |package|
+          dataset_uuids.include?(package.get("id"))
+        end
       end
 
       def diff_update(packages, datasets)
-        datasets = Hash[datasets.pluck(:uuid, :last_updated_at)]
+        datasets = Hash[datasets.pluck(:uuid, :updated_at)]
 
         packages.to_a.select do |package|
-          last_updated_at = datasets[package["id"]]
-          last_updated_at && package_is_changed?(package, last_updated_at)
+          updated_at = datasets[package.get("id")]
+          updated_at && package_is_changed?(package, updated_at)
         end
       end
 
       def diff_delete(packages, datasets)
-        package_uuids = packages.map { |package| package["id"] }
+        package_uuids = packages.map { |package| package.get("id") }
         datasets.where.not(uuid: package_uuids)
       end
 
-      def package_is_changed?(package, last_updated_at)
-        last_updated_at.iso8601 < package["metadata_modified"]
+      def package_is_changed?(package, updated_at)
+        updated_at.iso8601 < package.get("metadata_modified")
```

Comment:
Should this be `DateTime.parse(package.get("metadata_modified"))`? Our package is just a hash?

Reply:
It turns out `.parse` is quite expensive, whereas part of the beautify of ISO8601 is that the timestamps are lexicographically comparable, as here. I put it in a method to keep the optimisation separate.